ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/metrics/MetricRegistryOwner.scala

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.benchtool.metrics
+
+import scala.concurrent.duration.Duration
+import java.util.concurrent.TimeUnit
+import com.codahale.metrics.Slf4jReporter.LoggingLevel
+import com.codahale.metrics.{MetricRegistry, Reporter, Slf4jReporter}
+import com.daml.ledger.resources.{Resource, ResourceContext, ResourceOwner}
+import com.daml.metrics.MetricsReporter
+import com.daml.resources
+
+import scala.concurrent.Future
+
+class MetricRegistryOwner(
+    reporter: MetricsReporter,
+    reportingInterval: Duration,
+) extends ResourceOwner[MetricRegistry] {
+  override def acquire()(implicit
+      context: ResourceContext
+  ): resources.Resource[ResourceContext, MetricRegistry] = {
+    val registry = new MetricRegistry
+    for {
+      slf4JReporter <- acquire(newSlf4jReporter(registry))
+      _ <- acquire(reporter.register(registry))
+        .map(_.start(reportingInterval.toSeconds, TimeUnit.SECONDS))
+      // Trigger a report to the SLF4J logger on shutdown.
+      _ <- Resource(Future.successful(slf4JReporter))(reporter =>
+        Future.successful(reporter.report())
```

Comment:
can't you `reporter.report()` in the release closure when building the `slf4JReporter` on line 25?

Reply:
Now it looks like there are two resources involved instead of one

Reply:
Or something similar without nesting `Try` inside a `Future` but with recovery logic.

Reply:
import only `import com.daml.ledger.resources.Resource` btw

Reply:
Please correct me if I'm wrong but you don't call `close()` method on `Slf4jReporter` but only `report()`.

Reply:
Right, missed it. Updated

Reply:
But now in case `report()` fails we break graceful shutdown of the reporter. We can either add error recovery or stay with a separate resource for reporting. I'm still not sure which one is better :)

Reply:
Can `report()` throw?

Reply:
I didn't see how it can fail, but updated to handle it anyhow. My initial point is to handle this resource lifecycle in one go instead of splitting it in two.

Reply:
Ahh, you're right. I was trying to handle errors prematurely.
Thanks!